client/extensions/woocommerce/state/action-list/actions.js

```diff
@@ -0,0 +1,82 @@
+
+/**
+ * Internal dependencies
+ */
+import {
+	WOOCOMMERCE_ACTION_LIST_CREATE,
+	WOOCOMMERCE_ACTION_LIST_CLEAR,
+	WOOCOMMERCE_ACTION_LIST_STEP_START,
+	WOOCOMMERCE_ACTION_LIST_STEP_SUCCESS,
+	WOOCOMMERCE_ACTION_LIST_STEP_FAILURE,
+} from '../action-types';
+
+/**
+ * Action Creator: Create new Action List.
+ *
+ * @param {Object} actionList The sequential actionList to be carried out
+ * @return {Object} action
+ */
+export function actionListCreate( actionList ) {
+	return {
+		type: WOOCOMMERCE_ACTION_LIST_CREATE,
+		actionList,
+	};
+}
+
+/**
+ * Action Creator: Clear the current Action List.
+ * @return {Object} action
+ */
+export function actionListClear() {
+	return {
+		type: WOOCOMMERCE_ACTION_LIST_CLEAR,
+	};
+}
+
+/**
+ * Action Creator: Start action list step.
+ *
+ * @param {Number} [stepIndex] The index of the step. Defaults to zero (0), first index.
+ * @param {Date} [time=Date.now()] Optional timestamp.
+ * @return {Object} action
+ */
+export function actionListStepStart( stepIndex = 0, time = Date.now() ) {
+	return {
+		type: WOOCOMMERCE_ACTION_LIST_STEP_START,
+		stepIndex,
+		time,
+	};
+}
+
+/**
+ * Action Creator: Mark step as successful.
+ *
+ * @param {Number} stepIndex The index of the step.
+ * @param {Date} [time=Date.now()] Optional timestamp.
+ * @return {Object} action
+ */
+export function actionListStepSuccess( stepIndex, time = Date.now() ) {
```

Comment:
This action has to be dispatched by code using the action list. The action list code itself doesn't know how to execute individual steps like creating a product. The handler for each step has to ensure that step eventually results in dispatching `WOOCOMMERCE_ACTION_LIST_STEP_SUCCESS` or `WOOCOMMERCE_ACTION_LIST_STEP_FAILURE` to continue the action list execution or bring it to a conclusion.

See example here: https://github.com/Automattic/wp-calypso/pull/14559/files#diff-de56c6c7063f55776de904b3b2ebe1a7R83